libs/federation-util/src/Network/Federation/Util/Internal.hs

```diff
@@ -0,0 +1,68 @@
+module Network.Federation.Util.Internal where
+
+import Imports
+import Network.DNS.Resolver
+import Network.DNS.Types
+import Network.DNS
+import System.Random
+import Data.Text.Encoding (encodeUtf8)
+
+-- Given a prefix (e.g. _wire) and a domain (e.g. wire.com),
+-- provides a list of A(AAA) names and port numbers upon a successful
+-- DNS-SRV request, or `Nothing' if the DNS-SRV request failed.
+-- Modified version inspired from http://hackage.haskell.org/package/pontarius-xmpp
+srvLookup' :: Text -> Text -> ResolvSeed -> IO (Maybe [(Domain, Word16)])
+srvLookup' prefix realm resolvSeed = withResolver resolvSeed $ \resolver -> do
+    srvResult <- lookupSRV resolver $ encodeUtf8 $ prefix <> "._tcp." <> realm <> "."
+    case srvResult of
+        Right [] -> return Nothing
+        Right [(_, _, _, ".")] -> return Nothing -- explicit "not available"
```

Comment:
> Right [(_, _, _, ".")] -> return Nothing -- explicit "not available"

This is intended to match SRV records such as:

```
0 10 443 .
```

right? Can we add a test for that? Also, what happens if someone sets sth like:
```
0 10 443 .
0 10 443 bla.com.
```

Reply:
This case just deals with the case explicitly mentioned in [the SRV RFC](https://tools.ietf.org/html/rfc2782):

> If there is precisely one SRV RR, and its Target is "." (the root domain), abort.

If someone sets multiple records, one of them being the root domain, well, then just the normal logic applies and a server may not get contacted. We can't prevent people from misconfiguring their DNS.